### DISCUSSION TITLE
How can I add includes to `check_xxx` funcs based on previous checks?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1812] [Q&A]
https://github.com/xmake-io/xmake-repo/blob/master/packages/l/libintl/port/xmake.lua#L254

这里本来的逻辑是根据前面的HAVE_XLOCALE_H的结果来判断添加includes；但这里用`has_config("__HAVE_XLOCALE_H") and "xlocale.h" or "locale.h"`不起作用，所以只能用`is_plat("macosx")`来判断了。怎样实现根据前面check的结果来选择性的添加include？

**Comment by waruqi** [comment]
这种只能走 option + add_deps 控制下检测顺序，等待对应选项检测完，在当前 option的 after_check里面去 if opt:dep("xxz"):enabled() 判处理

因为option检测默认也是并行的

**Reply by waruqi** [reply]
参考这个 https://github.com/tboox/tbox/blob/122a479e626ee3fdd7d6c1117ec7c19212a1e087/src/xmake.lua#L60

**Reply by xq114** [reply]
那就是
```
option("__HAVE_NEWLOCALE")
	add_deps("__HAVE_XLOCALE_H")
	add_cfuncs("newlocale")
	before_check(function (option)
		if option:dep("__HAVE_XLOCALE_H"):enabled() then
			option:add("cincludes", "xlocale.h")
		else
			option:add("cincludes", "locale.h")
		end
	end)
```
这样写吗

**Reply by waruqi** [reply]
差不多吧，你先试试吧
